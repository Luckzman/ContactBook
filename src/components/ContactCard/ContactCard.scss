@import '../../styles/index.scss';

.card{
  width: 170px;
  min-width: 170px;
  box-shadow: 1px 1px 5px 1px map-get($colors, grey-color);
  border-radius: 20px;
  overflow: hidden;
  .card-details{
    width: 100%;
    min-height: 62px;
    margin-top: 10px;
    .name{
      margin: 5px auto;
      width: 150px;
      .word-break{
        word-break: break-all;
      }
    }
    .details{
      width: 150px;
      margin: 5px auto;
      font-style: italic;
      font-weight: normal;
      font-size: 12px;
      color: map-get($colors, light-grey);
      .word-break{
        word-break: break-all;
      }
    }
  }
  .card-footer{
    display: flex;
    width: 100%;
    margin-top: 10px;
    padding: 15px 0;
    justify-content: center;
    background-color: map-get($colors, faint-bg);
    border-top: 1px solid map-get($colors, nav-border);
    .footer-icon{
      display: flex;
      justify-content: space-around;
      width: 80%;
      .footer{
        position: relative;
        cursor: pointer;
        .icon{
          width: 18px;
          height: 18px;
          color: map-get($colors, brand-color);
          &:hover{
            ~ .icon-text{
              top: 9px;
            }
          }
        }
        .icon-text{
          position: absolute;
          top: 30px;
          right: 0;
          font-size: 10px;
          color: map-get($colors, brand-color );
          transition: all .3s;
        }
        .edit{
          right: 2px;
        }
      }
    }
  }
}