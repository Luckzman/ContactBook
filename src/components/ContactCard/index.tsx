import * as React from 'react';
import { ReactComponent as LikeIcon } from '../../assets/heart-regular.svg';
// import { ReactComponent as LikeSolidIcon } from '../../assets/heart-solid.svg';
import { ReactComponent as ShareIcon } from '../../assets/share-solid.svg';
import { ReactComponent as EditIcon } from '../../assets/user-edit-solid.svg';
import './ContactCard.scss';

const footerIcon = (Component: React.ReactElement, Text: string, styleText?: string): React.ReactElement => {
  return (
    <div className="footer">
      {Component}
      <p className={`icon-text ${styleText}`}>{Text}</p>
    </div>
  );
};

const likeIcon = <LikeIcon className="icon" />;
const shareIcon = <ShareIcon className="icon" />;
const editIcon = <EditIcon className="icon" />;

const ContactCard: React.FC = () => {
  return (
    <div className="card">
      <img className="img" src="https://res.cloudinary.com/dx0nauane/image/upload/v1576065923/avatar.jpg" alt="user" />
      <div className="card-details">
        <h4 className="name">
          <span className="word-break">Mary Ann</span>
        </h4>
        <p className="details">
          <span className="word-break">maryann@gmail.com</span>
        </p>
        <p className="details">
          <span className="word-break">07034222671</span>
        </p>
      </div>
      <div className="card-footer">
        {footerIcon(likeIcon, 'Like')}
        {footerIcon(shareIcon, 'Share', 'share')}
        {footerIcon(editIcon, 'Edit', 'edit')}
      </div>
    </div>
  );
};

export default ContactCard;
