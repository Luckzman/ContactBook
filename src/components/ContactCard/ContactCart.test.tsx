import React from 'react';
import { shallow } from 'enzyme';
import ContactCard from './';

it('renders without crashing', () => {
  const wrapper = shallow(<ContactCard />);
  expect(wrapper).toMatchSnapshot();
});
