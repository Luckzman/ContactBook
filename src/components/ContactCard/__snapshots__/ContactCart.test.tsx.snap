// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="card"
>
  <img
    alt="user"
    className="img"
    src="https://res.cloudinary.com/dx0nauane/image/upload/v1576065923/avatar.jpg"
  />
  <div
    className="card-details"
  >
    <h4
      className="name"
    >
      <span
        className="word-break"
      >
        Mary Ann
      </span>
    </h4>
    <p
      className="details"
    >
      <span
        className="word-break"
      >
        maryann@gmail.com
      </span>
    </p>
    <p
      className="details"
    >
      <span
        className="word-break"
      >
        07034222671
      </span>
    </p>
  </div>
  <div
    className="card-footer"
  >
    <div
      className="footer"
    >
      <ForwardRef(SvgHeartRegular)
        className="icon"
      />
      <p
        className="icon-text undefined"
      >
        Like
      </p>
    </div>
    <div
      className="footer"
    >
      <ForwardRef(SvgShareSolid)
        className="icon"
      />
      <p
        className="icon-text share"
      >
        Share
      </p>
    </div>
    <div
      className="footer"
    >
      <ForwardRef(SvgUserEditSolid)
        className="icon"
      />
      <p
        className="icon-text edit"
      >
        Edit
      </p>
    </div>
  </div>
</div>
`;
